presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -282,6 +277,24 @@ else if (type instanceof TinyintType
                 .orElseThrow(() -> new TableNotFoundException(tableName));
     }
 
+    public static HivePartition parsePartition(
+            SchemaTableName tableName,
+            String partitionId,
+            List<HiveColumnHandle> partitionColumns,
+            List<Type> partitionColumnTypes,
+            DateTimeZone timeZone)
+    {
+        List<String> keys = extractPartitionKeyValues(partitionId);
+        ImmutableMap.Builder<ColumnHandle, NullableValue> builder = ImmutableMap.builder();
+        for (int i = 0; i < partitionColumns.size(); i++) {
+            HiveColumnHandle column = partitionColumns.get(i);
+            NullableValue parsedValue = parsePartitionValue(partitionId, keys.get(i), partitionColumnTypes.get(i), timeZone);
```

Comment:
This code assumes that `partition_name` lists partition columns in proper order, however, `extractPartitionKeyValues` doesn't check for that; should we have an explicit check?

Reply:
The partitionName is something very confusing. Although all the column names are listed, and technically it is possible to shuffle the column names without loosing any information. However it is not legit. The partition values in the partition name should always be in order of partition columns.